crates/modelardb_compression/src/models/swing.rs

```diff
@@ -326,6 +350,26 @@ fn compute_slope_and_intercept(
     }
 }
 
+/// Computes numerator and denominator for Equation 6 
+/// in Section 3.2 of [Swing and Slide paper](https://dl.acm.org/doi/10.14778/1687627.1687645).
+/// It is a part of recording mechanism for finding the optimal slope with minimum MSE.
+fn compute_local_optimal_slope(
+    start_time: Timestamp,
+    first_value: f64,
+    end_time: Timestamp,
+    last_value: f64,
+) -> (f64, f64) {
+    if first_value == last_value || (first_value.is_nan() || last_value.is_nan()) {
```

Comment:
Unclear why this if expressions check if either of the values are `NaN` instead of both like `compute_slope_and_intercept()`. Maybe add a comment stating why or change the code to use `models::equal_or_nan()` if it should check if both values are `NaN`.